packages/simcore-sdk/src/simcore_sdk/node_ports_common/file_io_utils.py

```diff
@@ -327,31 +327,72 @@ async def _upload_file_part(
     raise exceptions.S3TransferError(msg)
 
 
-async def upload_file_to_presigned_links(
-    session: ClientSession,
-    file_upload_links: FileUploadSchema,
+def _get_file_size_and_name(
     file_to_upload: Path | UploadableFileObject,
-    *,
-    num_retries: int,
-    io_log_redirect_cb: LogRedirectCB | None,
-    progress_bar: ProgressBarData,
-) -> list[UploadedPart]:
-    file_size = 0
-    file_name = ""
+) -> tuple[int, str]:
     if isinstance(file_to_upload, Path):
         file_size = file_to_upload.stat().st_size
         file_name = file_to_upload.as_posix()
     else:
         file_size = file_to_upload.file_size
         file_name = file_to_upload.file_name
 
+    return file_size, file_name
+
+
+def _get_max_concurrency(file_to_upload: Path | UploadableFileObject) -> int:
     # NOTE: when the file object is already created it cannot be duplicated so
     # no concurrency is allowed in that case
-    max_concurrency = 4 if isinstance(file_to_upload, Path) else 1
+    return 4 if isinstance(file_to_upload, Path) else 1
+
+
+async def _process_batch(
+    *,
+    upload_tasks: list[Coroutine],
+    max_concurrency: int,
+    file_name: str,
+    file_size: int,
+    file_chunk_size: int,
+    last_chunk_size: int,
+) -> list[UploadedPart]:
+    results: list[UploadedPart] = []
+    try:
+        upload_results = await logged_gather(
+            *upload_tasks, log=_logger, max_concurrency=max_concurrency
+        )
+
+        for i, e_tag in upload_results:
+            results.append(UploadedPart(number=i + 1, e_tag=e_tag))
+    except ExtendedClientResponseError as e:
+        if e.status == web.HTTPBadRequest.status_code and "RequestTimeout" in e.body:
+            raise exceptions.AWSS3400RequestTimeOutError(e.body) from e
+    except ClientError as exc:
+        msg = (
+            f"Could not upload file {file_name} ({file_size=}, "
+            f"{file_chunk_size=}, {last_chunk_size=}):{exc}"
+        )
+        raise exceptions.S3TransferError(msg) from exc
+
+    return results
+
+
+async def upload_file_to_presigned_links(
+    session: ClientSession,
+    file_upload_links: FileUploadSchema,
+    file_to_upload: Path | UploadableFileObject,
+    *,
+    num_retries: int,
+    io_log_redirect_cb: LogRedirectCB | None,
+    progress_bar: ProgressBarData,
+) -> list[UploadedPart]:
+    file_size: int
+    file_name: str
```

Comment:
why is this needed? it is perfectly defined by the function ```_get_file_size_and_name```

Reply:
let's drop them